client/my-sites/checkout/checkout/index.jsx

```diff
@@ -343,6 +344,11 @@ export class Checkout extends React.Component {
 
 		let redirectTo = '/plans/';
 
+		if ( returnToBlockEditor ) {
+			page.redirect( `/block-editor/page/${ selectedSiteSlug }/home` );
+			return true;
+		}
```

Comment:
I'm wondering whether this branch should go _after_ the `if ( this.state.previousCare )` branch. I don't know what a "previous cart" is, but maybe there's a way of having nested carts? In which case it'd make sense for the user to back their way out of all the carts before finally emptying their last cart and being returned to the block editor.

But I don't know how it all works @niranjan-uma-shankar do you know what the `previousCart` state is for in this component?

Reply:
Oh, that's a good question. 

As far as I can see, [getExitCheckoutUrl](https://github.com/Automattic/wp-calypso/blob/master/client/lib/checkout/index.js#L71) checks what you had in your cart and redirects you to a related page, e.g., if `hasDomainMapping( cart )` then go to  `'/domains/add/mapping/'`.

So in this case we're _kinda_ doing the same thing, only outside of the method. I think.

Reply:
I haven't used previousCart before but from what I gather, it looks like the cart state from just before the cart was emptied. So at the point the cart is emptied, `this.props.cart`'s `product` prop is empty and `previousCart`'s product prop is used to determine the redirect destination(like in the example Ramon gave above). 

I think there's no harm in keeping the block-editor redirect where it is, but from keeping in line with the style maintained in that function, we could set the `redirectTo` value instead of using page.redirect in the if(returnToBlockEditor) block. We could also remove this `if` block and use `getExitCheckoutUrl`. 

Should this redirect work on the new checkout flow too? Or is that excluded for Gutenboarding? (maybe we have to change [here](https://github.com/Automattic/wp-calypso/blob/8c6ff289fc0b2750b4a22c1e0f266746e3b99fea/client/my-sites/checkout/composite-checkout/composite-checkout.js#L418) for the new checkout flow but I'm not entirely sure.)

Reply:
> I think there's no harm in keeping the block-editor redirect where it is, but from keeping in line with the style maintained in that function, we could set the redirectTo value instead of using page.redirect in the
if ( returnToBlockEditor ) { ... } block. We could also remove this if block and move things to getExitCheckoutUrl(in which case I think we don't need to check if previousCart exists). These are minor issues, so up to you if you'd like to do it.

Thanks! I'll make those changes. Good tips.

> Should this redirect work on the new checkout flow too? Or is that excluded for Gutenboarding? 

Another good question! 

I don't see anywhere in `shouldShowCompositeCheckout` that would [preclude Gutenboarding](shouldShowCompositeCheckout).

Is the new checkout live and working? I can't get it to load when I activate the A/B test, make my site USD and all the other conditions.

Maybe that can be another task.